File: Fuvekonse/services/general-service/internal/models/venue.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Venue is a global, reusable venue entity that can contain multiple locations.
type Venue struct {
	Id          uuid.UUID  `gorm:"type:uuid;primaryKey"`
	Name        string     `gorm:"type:varchar(255)"`
	Description string     `gorm:"type:varchar(1000)"`
	CreatedAt   time.Time  `gorm:"autoCreateTime"`
	ModifiedAt  time.Time  `gorm:"autoUpdateTime"`
	DeletedAt   *time.Time `gorm:"index"`
	IsDeleted   bool       `gorm:"default:false"`
	Locations   []Location `gorm:"foreignKey:VenueId"`
}

// Location is a global, reusable location belonging to a Venue.
type Location struct {
	Id          uuid.UUID  `gorm:"type:uuid;primaryKey"`
	VenueId     uuid.UUID  `gorm:"type:uuid;index"`
	Name        string     `gorm:"type:varchar(255)"`
	Description string     `gorm:"type:varchar(1000)"`
	Order       int        `gorm:"type:int;default:0"`
	CreatedAt   time.Time  `gorm:"autoCreateTime"`
	ModifiedAt  time.Time  `gorm:"autoUpdateTime"`
	DeletedAt   *time.Time `gorm:"index"`
	IsDeleted   bool       `gorm:"default:false"`
}

File: lib/features/schedule/domain/entities/venue.dart
import 'package:equatable/equatable.dart';

class VenueLocation extends Equatable {
  const VenueLocation({
    required this.id,
    required this.name,
    this.description = '',
  });

  final String id;
  final String name;
  final String description;

  @override
  List<Object?> get props => [id, name, description];
}

class Venue extends Equatable {
  const Venue({
    required this.id,
    required this.name,
    this.description = '',
    this.order = 0,
    this.mapX,
    this.mapY,
    this.locations = const [],
  });

  final String id;
  final String name;
  final String description;
  final int order;
  final double? mapX;
  final double? mapY;
  final List<VenueLocation> locations;

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        order,
        mapX,
        mapY,
        locations,
      ];
}
